renderer: add LUT-based implementation of isSymbol

jcollie opened the pull request:
The LUT-based lookup gives a ~20%-30% speedup over the "naive" isSymbol implementation.

<img width="1206" height="730" alt="Screenshot From 2025-09-04 22-45-10" src="https://github.com/user-attachments/assets/09a8ef3a-8b4b-43ba-963a-849338307251" />
<img width="1206" height="730" alt="Screenshot From 2025-09-04 22-41-54" src="https://github.com/user-attachments/assets/27962a88-f99c-446d-b986-30f526239ba3" />

Fixes #8523

jcollie committed:
add two LUT-based implementations of isSymbol

jcollie committed:
drop the new LUT type as no performance advantage detected

jcollie commented:
The benchmarks show two "tables" benchmarks. "tables1" is the original LUT implementation used for looking up unicode data. "tables2" is an alternate LUT implementation that I had hoped would do even better but turned out to have no benefits over the existing implementation.

mitchellh commented on src/unicode/lut.zig:
```diff
@@ -142,6 +142,32 @@ pub fn Tables(comptime Elem: type) type {
             return self.stage3[self.stage2[self.stage1[high] + low]];
         }
 
+        pub inline fn getInline(self: *const Self, cp: u21) Elem {
```
future note: There's probably little downside to marking `get` as inline and removing this. We don't need to do this in this PR because I don't want to cause more churn waiting on benchmarks. But this should be as simple as testing terminal throughput (we already have a benchmark).

jcollie replied:
I didn't include the results when I tested, but when I initially was doing some testing `inline` seemed to slow things down. I didn't test again after I fixed the problem with the lookups being optimized out.

jcollie replied:
Re-tested and `inline` gets a percent or so speedup.

mitchellh commented on src/unicode/symbols.zig:
```diff
@@ -0,0 +1,93 @@
+
+// This is not very fast in debug modes, so its commented by default.
+// IMPORTANT: UNCOMMENT THIS WHENEVER MAKING CHANGES.
+test "unicode symbols1: tables match ziglyph" {
```
Do we want to comment this out? Usually these slow tests we don't need to run every time. At the very least we should disable this under Valgrind.

mitchellh replied:
We should still disable during Valgrind, because Valgrind is at least 100,000x slower (not an exaggeration) and I don't think there's a practical Valgrind test here. There are other examples of how to do this.

jcollie replied:
Done.

mitchellh commented on src/unicode/lut.zig:
```diff
@@ -142,6 +142,32 @@ pub fn Tables(comptime Elem: type) type {
+            return self.stage3[self.stage2[self.stage1[high] + low]];
+        }
+
+        pub fn getBool(self: *const Self, cp: u21) bool {
```
Maybe blind, but actually are any of these used? Should we delete this?

mitchellh reviewed (changes requested):
Thanks for doing this work, some questions.

jcollie committed:
render: address review feedback

1. `inline` the table get.
2. Delete unused functions on the LUT table.
3. Disable the isSymbol test under valgrind
